panther_driver/src/panther_can.py

```diff
@@ -0,0 +1,173 @@
+#!/usr/bin/python3
+
+import canopen
+from dataclasses import dataclass
+from time import time, sleep
+from threading import Thread, Lock
+
+import rospy
+
+from constants import LEFT_WHEEL, RIGHT_WHEEL, VOLT_CHANNEL, AMP_CHANNEL
+
+
+@dataclass
+class MotorController:
+    wheel_pos = [0.0, 0.0]
+    wheel_vel = [0.0, 0.0]
+    wheel_curr = [0.0, 0.0]
+    battery_data = [0.0, 0.0] # V, I
+    runtime_stat_flag = [0, 0]
+    fault_flags = 0
+
+    def __init__(self, can_node_id, eds_file) -> None:
+        self.can_node = canopen.RemoteNode(can_node_id, eds_file)
+
+
+class PantherCAN:
+    def __init__(self, eds_file, can_interface) -> None:
+        self._can_net_err = False
+        self._max_err_per_sec = 2
+        self._err_times = [0] * self._max_err_per_sec
+        
+        self._lock = Lock()
+        self._network = canopen.Network()
+
+        self._motor_controllers = [
+            MotorController(1, eds_file),   # front
+            MotorController(2, eds_file)    # rear
+        ]
+        
+        self._network.connect(channel=can_interface, bustype='socketcan')
+
+        self._network.add_node(self._motor_controllers[0].can_node)
+        self._network.add_node(self._motor_controllers[1].can_node)
+        
+        self._connection_check_timer = Thread(target=self._can_net_check)  
+        self._connection_check_timer.start()
+
+        rospy.loginfo(f'[{rospy.get_name()}] Connected to the CAN bus.')
+    
+    def is_can_connection_correct(self):
+        return self._can_net_err
+
+    def set_wheels_enc_velocity(self, vel: list) -> None:
```

Comment:
```suggestion
    def write_wheels_enc_velocity(self, vel: list) -> None:
```

Same as with `querying`